### DISCUSSION TITLE
AttributeError: 'Train' object has no attribute 'lambda_delay'

### DISCUSSION THREAD

**Discussion by AmolMavuduru** [ourownstory/neural_prophet#225] [Q&A - get help using NeuralProphet]
I am running NeuralProphet on MacOS Catalina 10.15.5 and executed the following code in a Jupyter Notebook:

```
model = NeuralProphet(n_forecasts=3,
    n_lags=5,
    ar_sparsity=0.01,
    num_hidden_layers=2,
    yearly_seasonality=False,
    weekly_seasonality=False,
    daily_seasonality=False)

model.fit(sp500_data, 
          freq='D',
          epochs=100)
```

I got an error titled "AttributeError: 'Train' object has no attribute 'lambda_delay'". Here is the full stack trace:

---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-7-085265468e66> in <module>
      9 model.fit(sp500_data, 
     10           freq='D',
---> 11           epochs=100)

~/opt/anaconda3/lib/python3.7/site-packages/neuralprophet/forecaster.py in fit(self, df, freq, epochs, validate_each_epoch, valid_p, use_tqdm, plot_live_loss)
    753             metrics_df = self._train(df_train, df_val, use_tqdm=use_tqdm, plot_live_loss=plot_live_loss)
    754         else:
--> 755             metrics_df = self._train(df, use_tqdm=use_tqdm, plot_live_loss=plot_live_loss)
    756         if epochs is not None:
    757             self.config_train.epochs = default_epochs

~/opt/anaconda3/lib/python3.7/site-packages/neuralprophet/forecaster.py in _train(self, df, df_val, use_tqdm, plot_live_loss)
    622             if val:
    623                 val_metrics.reset()
--> 624             epoch_metrics = self._train_epoch(e, loader)
    625             metrics_live["{}".format(list(epoch_metrics)[0])] = epoch_metrics[list(epoch_metrics)[0]]
    626             if val:

~/opt/anaconda3/lib/python3.7/site-packages/neuralprophet/forecaster.py in _train_epoch(self, e, loader)
    483         if self.n_lags > 0:  # slowly increase regularization until lambda_delay epoch
    484             reg_lambda_ar = utils.get_regularization_lambda(
--> 485                 self.config_train.ar_sparsity, self.config_train.lambda_delay, e
    486             )
    487         for inputs, targets in loader:

AttributeError: 'Train' object has no attribute 'lambda_delay'

**Comment by ourownstory** [comment]
Hi @AmolMavuduru I am sorry, this was a bug in an earlier version on pip. It should have been resolved in 0.2.7.

To resolve this, you can directly install from github:
```shell
git clone <copied link from github>
cd neural_prophet
pip install .
```

**Reply by AmolMavuduru** [reply]
Thanks for your response! I installed NeuralProphet directly from GitHub and I see that when I include the number of epochs in my model definition, the code works fine, otherwise, the previous AttributeError occurs again. But for now, the issue is fixed.

**Reply by ourownstory** [reply]
@AmolMavuduru that should not be the case. Then there is either still a bug, or your venv might be using an older version. 
Can you please share if you re-installed it in a new virtual environment, and what version pip shows you? `pip list | grep neuralprophet`
